adapters/nativery/nativery.go

```diff
@@ -0,0 +1,302 @@
+package nativery
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"maps"
+	"net/http"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v3/adapters"
+	"github.com/prebid/prebid-server/v3/config"
+	"github.com/prebid/prebid-server/v3/metrics"
+	"github.com/prebid/prebid-server/v3/openrtb_ext"
+)
+
+// Function used to  builds a new instance of the Nativery adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	// build bidder
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+// makeRequests creates HTTP requests for a given BidRequest and adapter configuration.
+// It generates requests for each ad exchange targeted by the BidRequest,
+// serializes the BidRequest into the request body, and sets the appropriate
+// HTTP headers and other parameters.
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	reqCopy := *request
+	var errs []error
+
+	// check if the request come from AMP
+	var isAMP int
+	if reqInfo.PbsEntryPoint == metrics.ReqTypeAMP {
+		isAMP = 1
+	}
+
+	var widgetId string
+
+	// attach body request for all the impressions
+	validImps := []openrtb2.Imp{}
+	for i, imp := range request.Imp {
+		reqCopy.Imp = []openrtb2.Imp{imp}
+
+		nativeryExt, err := buildNativeryExt(&reqCopy.Imp[0])
```

Comment:
@andreafassina :  I could see `imp` object is used by 
1. buildNativeryExt 
2. buildRequest

So, code can be refactored and could be more readable like
```go
for i, imp := range request.Imp {

   nativeryExt, err := buildNativeryExt(&imp)
   ...
   reqCopy.Imp = []openrtb2.Imp{imp}
   buildRequest(reqCopy, nativeryExt)
   ...
}
```